src/azure_storage_account_client.cpp

```diff
@@ -78,9 +79,17 @@ static std::string AccountUrl(const KeyValueSecret &secret, const std::string &p
 }
 
 static Azure::Storage::Blobs::BlobClientOptions
-ToBlobClientOptions(const Azure::Core::Http::Policies::TransportOptions &transport_options) {
+ToBlobClientOptions(const Azure::Core::Http::Policies::TransportOptions &transport_options,
+                    std::shared_ptr<HTTPState> http_state) {
 	Azure::Storage::Blobs::BlobClientOptions options;
 	options.Transport = transport_options;
+	if (nullptr != http_state) {
+		// Because we mainly want to have stats on what has been needed and not on
+		// what has been used on the network, we register the policy on `PerOperationPolicies`
+		// part and not the `PerRetryPolicies`. Network issues will result in retry that can
```

Comment:
Hum I understand but I have mixing feeling about this. Also it can make tests failed if a retry happen (I agree it will probably never happen if we stay in local with azurite)

Reply:
I guess the test stability argument is fair, lets leave it as is then!